# Acquire

This file contains my data acquisition in details

**Imports**

In [19]:
# importing my data
import pandas as pd
import numpy as np

# import sys to use system functions
# this allow for imports form directories different from the current
import sys
sys.path.append("/Users/migashane/CodeUp/Data_Science/telco_churn_classification_project/util_")
import acquire_

**Get data**

I will use the `telco` churn data from the Codeup databse

In [ ]:
query = """
    SELECT *
    FROM customers #payment_types
    JOIN contract_types ct USING(contract_type_id)
    JOIN internet_service_types ist USING(internet_service_type_id)
    JOIN payment_types pt USING(payment_type_id);
    """

# get telco data from codeup database
telco, query = acquire_.get_codeup_sql_data_(db_name="telco_churn", query=query)
telco.head()

- The `query` is returned here for future reference on the exact data I pulled from codeup database


## Understand my data

In [33]:
# look at the shape of my data
telco.shape

(7043, 24)

In [36]:
# see what columns I have in my dataframe
telco.columns

Index(['payment_type_id', 'internet_service_type_id', 'contract_type_id',
       'customer_id', 'gender', 'senior_citizen', 'partner', 'dependents',
       'tenure', 'phone_service', 'multiple_lines', 'online_security',
       'online_backup', 'device_protection', 'tech_support', 'streaming_tv',
       'streaming_movies', 'paperless_billing', 'monthly_charges',
       'total_charges', 'churn', 'contract_type', 'internet_service_type',
       'payment_type'],
      dtype='object')

In [34]:
# look at the information contained my data
telco.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7043 entries, 0 to 7042
Data columns (total 24 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   payment_type_id           7043 non-null   int64  
 1   internet_service_type_id  7043 non-null   int64  
 2   contract_type_id          7043 non-null   int64  
 3   customer_id               7043 non-null   object 
 4   gender                    7043 non-null   object 
 5   senior_citizen            7043 non-null   int64  
 6   partner                   7043 non-null   object 
 7   dependents                7043 non-null   object 
 8   tenure                    7043 non-null   int64  
 9   phone_service             7043 non-null   object 
 10  multiple_lines            7043 non-null   object 
 11  online_security           7043 non-null   object 
 12  online_backup             7043 non-null   object 
 13  device_protection         7043 non-null   object 
 14  tech_sup

In [47]:
# count of numeric columns and object columns
numerics = len(telco.select_dtypes("number").columns)
objects = len(telco.select_dtypes("object").columns)

numerics, objects

(6, 18)

In [48]:
# get discriptive statistics of my data
telco.describe()

payment_type_id  internet_service_type_id  contract_type_id  \
count      7043.000000               7043.000000       7043.000000   
mean          2.315633                  1.872923          1.690473   
std           1.148907                  0.737796          0.833755   
min           1.000000                  1.000000          1.000000   
25%           1.000000                  1.000000          1.000000   
50%           2.000000                  2.000000          1.000000   
75%           3.000000                  2.000000          2.000000   
max           4.000000                  3.000000          3.000000   

       senior_citizen       tenure  monthly_charges  
count     7043.000000  7043.000000      7043.000000  
mean         0.162147    32.371149        64.761692  
std          0.368612    24.559481        30.090047  
min          0.000000     0.000000        18.250000  
25%          0.000000     9.000000        35.500000  
50%          0.000000    29.000000        70.350000  
75%          0.000000    55.000000        89.850000  
max          1.000000    72.000000       118.750000

**What I see:**

- I have 7043 rows and 24columns
- 6 of the 24 columns are numric while 18 of them are string object colums
- out of the 6 numeric columns 1 is a float colum
- I have no null values in my entire data
- I also see the descriptive statistics of my data
